nix/nixos/cardano-explorer-service.nix

```diff
@@ -0,0 +1,106 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let
+  cfg = config.services.cardano-explorer;
+  inherit (import ../.. { }) cardano-explorer;
+in {
+  config = mkIf cfg.enable (let stateDirBase = "/var/lib/";
+  in {
+    networking.firewall.allowedTCPPorts = [ 3001 8100 ];
+
+    users.groups.${cfg.group}.gid = 10017;
+
+    users.users.${cfg.user} = {
+      description = "cardano-explorer daemon user";
+      uid = 10017;
+      group = cfg.group;
+    };
+
+    systemd.services = {
+      cardano-explorer-node = {
+        description = "cardano-explorer node";
+        wantedBy = [ "multi-user.target" ];
+        after = [ "postgresql.service" ];
+        preStart = ''
+          for i in {1..${toString cfg.secondsToWaitForSocket}};
+            do test -S ${cfg.socketPath} && break || sleep 1;
+          done
+
+          chgrp ${cfg.group} ${config.services.cardano-exporter.socketPath}
+          chmod g+w ${config.services.cardano-exporter.socketPath}
+        '';
+        serviceConfig = {
+          PermissionsStartOnly = "true";
+          User = cfg.user;
+          Group = cfg.group;
+          StateDirectory = removePrefix stateDirBase cfg.stateDir;
+          WorkingDirectory = cfg.stateDir;
+        };
+        environment.PGPASSFILE = __toFile "pgpass" "/tmp:5432:cexplorer:*:*";
+        script = ''
+          ${cardano-explorer}/bin/cardano-explorer
```

Comment:
this service conflicts with `systemd.services.cardano-exporter`
the name implies it does the same thing, but the .script does the same thing as `cardano-explorer-webapi`